react-native-reanimated.d.ts

```diff
@@ -210,15 +210,12 @@ declare module 'react-native-reanimated' {
     };
 
     export type AnimateProps<
-      S extends object,
-      P extends {
-        style?: StyleProp<S>;
-      }
+      P extends object
     > = {
       [K in keyof P]: K extends 'style'
-        ? StyleProp<AnimateStyle<S>>
+        ? StyleProp<AnimateStyle<P[K]>>
         : P[K] | AnimatedNode<P[K]>;
-    };
+    } & { animatedProps?: AnimateProps<P> };
```

Comment:
BTW, `AnimateProps<P>` here shouldn't be `Omit<AnimateProps<P>, "animatedProps">`?

Reply:
Could you update the test for that? I'm not sure what you mean.

Reply:
I meant that in the current form we can pass `{animatedProps: {animatedProps: {}}}` if I'm not mistaken.